### PR TITLE
Bootstrap ShapeIt Github Bot

### LINKED ISSUE #34: Setup GitHub App
Related to #27

### PR DESCRIPTION
**Pull request by Amzani**
### Pitch onboarding

1. Create a [new Pitch](https://github.com/shapeitapp/shapeitapp/issues/new?assignees=&labels=Pitch&projects=&template=pitch.yml&title=%5BPITCH%5D%3A)
2. Add the pitch in a ShapeIt project board (example: https://github.com/orgs/shapeitapp/projects/2)

### Scope onboarding

1. Add a scope as usual to a Pitch by using the `related to #NUMBER`
2. An onboarding comment should appear on the scope

> Note: If `related to` is absent in the scope no onboarding comment is shown for now. 

### Scope progress

1. Comment `/progress XX` ... as usual
2. The initial scope comment gets updated
3. The original progress comment get minimised (not deleted for now)

Demo: https://github.com/shapeitapp/shapeitapp/issues/28

### Some details

- Example https://github.com/shapeitapp/shapeitapp/issues/55
- The Github app is for now hosted in my private server until we find an alternative.
- App Config URL: https://github.com/organizations/shapeitapp/settings/apps/shapeit-bot
- App install URL: https://github.com/apps/shapeit-bot

Fixes https://github.com/shapeitapp/shapeitapp/issues/34 https://github.com/shapeitapp/shapeitapp/issues/28 https://github.com/shapeitapp/shapeitapp/issues/25

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,59 +1,61 @@
...
+    if (description) {
+      await addScopeToBet(context, description)
+      await handleNewScope(context, owner, repo, issueNodeId, issueNumber)
+    }
   })
 
-  // For more information on building apps:
-  // https://probot.github.io/docs/
+  app.on("projects_v2_item.edited", async (context) => {
+    console.log(`projects_v2_item.edited triggered`)
```
Comment (fmvilas) [inline index.js]:
```suggestion
```

File: index.js [hunk 2]
```diff
@@ -1,59 +1,61 @@
...
+    if (itemType === "Issue") {
+      const item = await getProjectNodeItem(context, projectNodeId, itemNodeId)
+      const pitchType = item.fieldValues.nodes.find(item => item?.field?.name === KIND_FIELD && item?.name === PITCH_KIND_LABEL)
+      const betType = item.fieldValues.nodes.find(item => item?.field?.name === KIND_FIELD && item?.name === BET_KIND_LABEL)
+      const owner = item.content.repository.owner.login
+      const repository = item.content.repository.name
+      const issueNodeId = item.content.id
+      const issueNumber = item.content.number
+      if (pitchType || betType) {
+        console.log("That's a pitch or a Bet")
```
Comment (fmvilas) [inline index.js]:
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -1,59 +1,61 @@
...
+        await handleNewPitch(context, owner, repository, issueNodeId, issueNumber)
+      }
+    } else {
+      console.log("Not an issue")
+    }
```
Comment (fmvilas) [inline index.js]:
```suggestion
    }
```

File: index.js [hunk 4]
```diff
@@ -1,59 +1,61 @@
...
+  })
 
-  // To get your app running against GitHub, see:
-  // https://probot.github.io/docs/development/
+  app.on("issue_comment.created", async(context) => {
+    console.log(`issue_comment.created`)
```
Comment (fmvilas) [inline index.js]:
```suggestion
```

File: lib/onboarding.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+**Option 1**
+1. Edit the Pitch or Bet issue
+2. Add your scope under the scope section
+
+See this [example](https://github.com/asyncapi/studio/issues/748)
+
+**Option 2**
+1. Create a new issue
+2. Add this keywork in the description \`related to #ISSUE_BET_NUMBER\`
```
Comment (fmvilas) [inline lib/onboarding.js:19]:
```suggestion
2. Add this keyword in the description \`related to #ISSUE_BET_NUMBER\`
```

File: lib/onboarding.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  {
+    projectNodeId: projectNodeId
+  })
+  const items =  data.node.items.nodes
+  const nodeItem = items.find(item => item.id === itemNodeId)
+  return nodeItem
+}
+
+const handleNewScope = async(context, owner, repo, issueNodeId, issueNumber) => {
+  console.log("Scope onboarding")
```
Comment (fmvilas) [inline lib/onboarding.js]:
```suggestion
```

File: lib/onboarding.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  const commentHeader = 'scope'
+  const commentText = `Thanks for creating a new scope 🥳. You can now communicate your progress
+To do so, you have to leave a comment in the issue or pull request with the following syntax::
```
Comment (fmvilas) [inline lib/onboarding.js]:
```suggestion
To do so, you have to leave a comment in this issue or pull request with the following syntax:
```

File: lib/onboarding.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+/progress 50
+A few notes:
+* We got this figured out :tada:
+* We\'re going to use [this library](#link-to-website) to avoid losing time implementing this algorithm.
+* We decided to go for the quickest solution and will improve it if we got time at the end of the cycle.
+\`\`\`
+    `
+  if (isScope(context.payload.issue.body)) {
+    await addComment(context, owner, repo, issueNodeId, issueNumber, commentHeader, commentText)
+    console.log("Scope Comment added")
```
Comment (fmvilas) [inline lib/onboarding.js]:
```suggestion
```

File: lib/progress.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
+const { findPreviousComment, headerComment, updateComment, minimizeComment } = require('./comments')
+
+//@samz updated progress to 60% Screenshot 2023-07-05 at 11 51 11 on June 30, 2023 (and the description below if any)
+
+const handleProgress = async(context, owner, repo, issueNumber) => {
+    console.log("Handle progress command")
+    // const searchComment = await findPreviousComment(context, owner, repo, issueNumber, '/progress', false)
```
Comment (fmvilas) [inline lib/progress.js]:
```suggestion
```

File: lib/progress.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    const comment = context.payload.comment
+    if (comment.body.includes('/progress') && !comment.user.login.includes('-bot')) {
```
Comment (fmvilas) [inline lib/progress.js]:
```suggestion
    if (comment.body.includes('/progress') && !comment.user.login.endsWith('-bot')) {
```

File: lib/scopes.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
+const { trackedIssuesQuery, updateIssueMutation } = require('./queries')
+
+const parseIssueDescription = async(context) => {
+    const regexSimple = /(Related\sto\s+(\S+)?#(\d+))/
```
Comment (fmvilas) [inline lib/scopes.js]: This won't match URLs, just `#ISSUE_NUMBER`, which is great but sometimes the Pitch or Bet is in another repo so we need to be able to understand full URLs here too.

File: lib/scopes.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    const description = context.payload.issue.body
+    const match = regexSimple.exec(description)
+    if (!match) {
+      console.log("Nothing to do in this issue")
```
Comment (fmvilas) [inline lib/scopes.js]:
```suggestion
```

**Review by fmvilas — CHANGES_REQUESTED** [review]
Super pumped this is coming! 🙌 

Left some comments but it's mostly about cleaning console.logs 😄

**Comment by Amzani** [comment]
@fmvilas thanks for your review, I've fixed a couple of stuffs

**Comment by Amzani** [comment]
@fmvilas added deployment to fly.io / you should get an invitation by email.

**Review by fmvilas — APPROVED** [review]

**Pull request merged**